packages/okidoc-md/src/buildDocumentationSource/buildDocumentationSourceAST/buildDocumentationSourceAST.js

```diff
@@ -24,38 +25,66 @@ function buildDocumentationSourceAST({
     createApiFunction,
   } = getVisitorApi({ tag, visitorPath });
 
-  const interfaces = [];
-  const classDeclarations = [];
-  const classBodyItems = [];
-  const functions = [];
-
   // NOTE: read about visitors
   // https://github.com/thejameskyle/babel-handbook/blob/master/translations/en/plugin-handbook.md#toc-visitors
   // https://github.com/babel/babel/blob/master/packages/babel-traverse/src/visitors.js
-  const visitors = {
-    'InterfaceDeclaration|TSInterfaceDeclaration': (path, state) => {
-      const interfaceDeclaration = createApiInterface(path.node, path);
+  const interfaceVisitor = {
+    'GenericTypeAnnotation|TSTypeReference': (path, state) => {
+      const file = state.dependenciesTree.get(state.filePath);
+
+      if (file) {
+        // flow support with GenericTypeAnnotation, how to verify?
+        const typeName = t.isTSTypeReference(path.node) ? path.node.typeName.name : path.node.id.name;
+        // NOTE: it might be the case when more than 1 import exists.
+        // Example:
+        // import { MyType } from './types';
+        // import { MyType } from './another-type';
+        // in this case both of them would be added to the doc
+        file.dependencies.forEach((importSpecifiers, absoluteFilePath) => {
+          const importSpecifier = importSpecifiers.find(specifier => specifier.local.name === typeName);
+
+          if (importSpecifier) {
+            const interfaceFile = state.dependenciesTree.get(absoluteFilePath);
+
+            if (interfaceFile) {
+              const importName = t.isImportDefaultSpecifier(importSpecifier) ?
+                importSpecifier.local.name :
+                importSpecifier.imported.name;
+              const interfaceNode = interfaceFile.interfaces.find(node => node.id.name === importName);
 
-      interfaces.push(interfaceDeclaration);
-      globalInterfaces[
-        `${state.filePath}.${path.node.id.name}`
-      ] = interfaceDeclaration;
+              if (interfaceNode) {
+                state.interfaces.push(
+                  createApiInterface(interfaceNode)
+                );
+              }
+            }
+          }
+        });
+      }
     },
-    ...createApiVisitor((path, options) => {
+  };
+
+  const traverseInterfaces = (path, state) => path.traverse(interfaceVisitor, state);
+
+  const visitors = {
+    ...createApiVisitor((path, options, state) => {
       if (path.isClassDeclaration()) {
-        classDeclarations.push(
+        state.classDeclarations.push(
           createApiClassDeclaration(path.node, path, options),
         );
+        traverseInterfaces(path, state);
```

Comment:
mb better to traverse a node, returned by `createApi[ClassDeclaration|ClassMethod|..]`?